### DISCUSSION TITLE
parcel 2.0 web server can not find index.html page

### DISCUSSION THREAD

**Discussion by crazyit** [parcel-bundler/parcel#7456] [Q&A]
Excuse me, my project is a web video game, which is mainly coded in typescript. All the typescript code will be compiled into JavaScript, I use an index.html, embedded an index.js, the index.js file loads all the compiled javascript code from Typesript.I use parcel version 2.0 and execute the parcel command in the src directory of the project, "parcel . /src/Main.ts --dist-dir . /bin/dist --public-url . /", main.ts is the web video game entry file. parcel successfully packaged all the Typescripte code to bin/dist directory, but when I load localhost:1234 in the browser, the browser prompt page not found, I don't know what is the reason, please help me. Thanks. My project directory is as follows
![Uploading Snipaste_2021-12-16_22-48-01.jpg…]()
.

**Comment by crazyit** [comment]
![Snipaste_2021-12-16_23-06-43](https://user-images.githubusercontent.com/6240481/146396548-3d2cd915-0f0c-4dab-a812-90da348b370e.jpg)
![Snipaste_2021-12-16_22-48-01](https://user-images.githubusercontent.com/6240481/146396332-79d54f66-5c0a-44e9-9931-41ca42a76d0b.jpg)![Snipaste_2021-12-16_22-48-36](https://user-images.githubusercontent.com/6240481/146396415-aa1727ec-6feb-4418-a5ca-97af2fd988ee.jpg)

**Comment by crazyit** [comment]
![image](https://user-images.githubusercontent.com/6240481/146737000-3479e6c1-d9d7-4c94-8e18-be18dcc26909.png)

**Comment by Shinyaigeek** [comment]
At first glance, the application you are building looks like it works in a web browser. If so, you should specify `.html` file as a build entry point.

ex)

**Main.ts**
```ts
// some code
```

**index.html**
```html
<html>
  <head></head>
  <body>
    <script type=“module” src=“./src/Main.ts”></script>
  </body>
</html>
```

**serve command on cli**

```sh
parcel ./index.html
```


If the above solution doesn't work, please comment with a slightly more detailed report (reproduction repository, etc.)

**Reply by crazyit** [reply]
Thank you vAccording to the method you provided, I modified the project entry and structure , and I can find the entry file, but the startup errorery much for your reply.
![image](https://user-images.githubusercontent.com/6240481/146770955-7620fc86-f905-41ac-8468-aecf9c5a7dcb.png)

**Reply by crazyit** [reply]
Why can't I find index.html if I don't change the project structure and entrance?

**Reply by Shinyaigeek** [reply]
This error `Uncaught SyntaxError: Unexpected token <` seems to be caused because Browser requests JavaScript file which does not exist so parcel dev server returns `index.html` instead and Browser tries to parse `index.html` as JavaScript file.

Can you share the minimum reproduction repository? (at a glance, I think `loadScript` function in your HTML causes this problem. Mabe, you can fix this issue by using `<script />` normally, instead of using `loadScript`)

**Reply by crazyit** [reply]
Thank you very much for your reply, it took you a while. I've uploaded the demo project at https://github.com/crazyit/parcelDemo

**Reply by Shinyaigeek** [reply]
Thank you for sharing, I watched it.

I pulled your repository and try to execute your shared command `parcel . /index.html --dist-dir . /bin/dist --public-url . /` (I replace `src/Main.ts` with `index.html`), but I cannot reproduce your issue due to 404 error, not Unexpected token error. Can you share more detailed reproduction info such as your command. (Your shared project seems to build project with web pack, but do you intend to bundle webpack’s output with parcel?)

Btw, I think it is the cause of this issue that the `loadScript` module isn't passed the correct URL. it will be solved if you load the script with the `<script />` tag and entrust the resolution of the dependency to parcel, or pass the correct URL as an argument to `loadScript`.

**Reply by crazyit** [reply]
Much appreciated, I'll try changing the call to loadscript parameter. But the command I was using before was parcel . /index.html, not parcel . /index.html --dist-dir . /bin/dist --public-url . /. I was using webpack for my project before, but webpack build was too slow. So I want to switch to parcel packaging

**Reply by Shinyaigeek** [reply]
Oh sorry, this was my typo.

you are trying to dynamically load index.js etc in `bin/` with the `loadScript` module, but parcel cannot resolve the dependency through the `loadScriptP` module, and `parcel serve` does not have the function to serve all files under this directory, so I think it is necessary to prepare a static dev server for serving `bin/`

**Reply by crazyit** [reply]
Thank you very, very much, my game project recently to catch up, did not have time to read your reply, sorry. I will try the solution you mentioned, thank you!
